packages/transformers/js/src/visitors/dependencies.js

```diff
@@ -15,13 +15,13 @@ export default {
   ExportNamedDeclaration(node, asset) {
     asset.meta.isES6Module = true;
     if (node.source) {
-      addDependency(asset, node.source);
+      addDependency(asset, node.source, {isWeak: true});
```

Comment:
Can this change affect bundling without scope hoisting?

Reply:
I think this might cause unexpected results when not using scope hoisting. The logic to defer transformation is based on the symbol metadata, which is only collected in the scope hoisting visitor. So with node symbol metadata it would look like nothing is being used, and you could get to the end and some code would be excluded that shouldn't be. Might be worth testing against `lodash-es` or something.

Reply:
I've made `isWeak` have 3 values now: true, false, unknown (null/undefined); that's cleaner at any rate.